errors/testing.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2022 Mineiros GmbH
```

Comment:
The idea of exporting some test facilities is really cool, Im just not so sure if we should do it in the same package/namespace of the actual code, like when you see the package docs there will also be a bunch of assertion functions there...maybe it is OK and it is just some subjective bias because I never saw it before, but could make it more annoying for people who are just looking to use the errors API and not actually write tests (although they should write tests too :laughing: ).

Maybe we could go with "test/errors" like we have for other testing facilities on the project.

Reply:
yeah I thought the same thing but I didn't know it was worth the effort. Will do.